File: Bridge/DrawingWithIoCAutoFac/Circle.cs
namespace DrawingWithIoCAutoFac
{
    public class Circle : Shape
    {
        private float _radius;

        public Circle(IRenderer renderer, float radius) : base(renderer)
        {
            _radius = radius;
        }

        public override void Draw()
        {
            _renderer.RenderCircle(_radius);
        }

        public override void Resize(float factor)
        {
            _radius *= factor;
        }
    }
}

File: Python/Bridge/Renderer/circle.py
from renderer import Renderer
from shape import Shape


class Circle(Shape):
    def __init__(self, renderer: Renderer, radius: int) -> None:
        super().__init__(renderer)
        self.radius = radius

    def draw(self):
        self.renderer.renderer_circle(self.radius)

    def resize(self, factor):
        self.radius *= factor
